### PR TITLE
fix: enable new post emails for freeform posts

### PR DESCRIPTION
**Pull request by rebelchris**
Enabling post_mention and squad_post_added notifications emails for freeform as well 🤯

### CODE REVIEW

File: src/common/twitter.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
 import { Post } from '../entity';
 
+export const truncateToTweet = (text: string): string => {
```
Comment (capJavert) [inline src/common/twitter.ts:3]:
```suggestion
export const truncateToTweet = (text?: string): string => {
```

File: src/workers/newNotificationV2Mail.ts [hunk 2]
```diff
@@ -448,32 +470,51 @@ const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
...
       post_link: addNotificationEmailUtm(
         notification.targetUrl,
         notification.type,
       ),
+      user_reputation: author.reputation,
+    };
+
+    if (!sharedPost) {
+      // Freeform post
+      return {
```
Comment (capJavert) [inline src/workers/newNotificationV2Mail.ts:507]: Is this right? Regular posts also don't have shared post?

Comment (rebelchris) [reply src/workers/newNotificationV2Mail.ts:507]: What do you consider a regular post? just article post?
You can't be mentioned on a regular post so it won't come to this part.

Comment (capJavert) [reply src/workers/newNotificationV2Mail.ts:507]: By mentioned you mean in post's content?

**Review by capJavert — APPROVED** [review]
Awesome!

**Pull request merged**